ui/app/src/components/DateTimePicker.tsx

```diff
@@ -95,80 +128,124 @@ interface timezoneType {
-    timezone = moment.tz.guess(),
-    dateFormat = 'YYYY-MM-DD',
-    timeFormat = 'HH:MM:ss',
-    controls = ['date', 'time', 'timezone']
-  } = props;
+const getDateTimeMoment = (dateString, timezoneObj) => {
+  let dateMoment;
 
-  let initialDateMoment;
-
-  switch(typeof initialDate) {
+  switch (typeof dateString) {
     case 'string':
     case 'number':
-      initialDateMoment = moment(initialDate);
+      dateMoment = timezoneObj ? moment.tz(dateString, timezoneObj.timezoneName) : moment.tz(dateString);
       break;
     case 'object':
       // moment object, stays the same
-      initialDateMoment = initialDate;
+      dateMoment = dateString;
       break;
     default:
-      initialDateMoment = moment();
+      dateMoment = moment();
   }
 
-  const timezoneObj = timezones.find( tz => (tz.timezoneName === timezone) );
-  const [ selectedDateTime, setSelectedDateTime ] = useState(initialDateMoment);
-  const [ selectedTimezone, setSelectedTimezone ] = useState(timezoneObj);
+  return dateMoment;
+};
 
-  const handleDateChange = (name: string) => (date: Date | null) => {
-    let updatedDateTime = selectedDateTime;
+const DateTimePicker = withStyles(dateTimePickerStyles)((props: DateTimePickerProps) => {
+  const {
+    id,
+    classes,
+    onChange,
+    onError,
+    date = moment(),
+    disabled = false,
+    controls = ['date', 'time', 'timezone'],
+    datePickerProps = {},
+    timePickerProps = {},
+    timeZonePickerProps = {}
+  } = props;
+  const [pickerState, setPickerState] = useState({
+    dateValid: true,
+    timeValid: true,
+    timezoneValid: true
+  });
+
+  // Set defaults on wrapped controls props
+  if (nou(timeZonePickerProps.timezone)) {
+    timeZonePickerProps['timezone'] = moment.tz.guess();
+  }
+  if (nou(datePickerProps.dateFormat)) {
+    datePickerProps['dateFormat'] = 'YYYY-MM-DD';
+  }
+  if (nou(datePickerProps.disablePast)) {
+    datePickerProps['disablePast'] = false;
+  }
+  if (nou(timePickerProps.timeFormat)) {
+    timePickerProps['timeFormat'] = 'HH:mm';
+  }
+
+  let dateMoment;
+  let timeMoment;
+  let timezoneObj = timezones.find(tz => (tz.timezoneName === unescape(timeZonePickerProps.timezone)));
+  dateMoment = getDateTimeMoment(date, timezoneObj);
+  timeMoment = getDateTimeMoment(date, timezoneObj);
+  const { formatMessage } = useIntl();
+
+  const handleDateChange = (name: string) => (newDate: Date | null) => {
+    let updatedDateTime = moment(newDate);
 
     switch (name) {
       case 'scheduledDate':
-        updatedDateTime.date(date.getDate());
-        updatedDateTime.month(date.getMonth());
-        updatedDateTime.year(date.getFullYear());
+        if (updatedDateTime._isValid) {
+          updatedDateTime.hours(timeMoment.toDate().getHours());
+          updatedDateTime.minutes(timeMoment.toDate().getMinutes());
+          updatedDateTime.seconds(timeMoment.toDate().getSeconds());
+          updatedDateTime.milliseconds(timeMoment.toDate().getSeconds());
```

Comment:
Copy paste issue (getSeconds to set milliseconds).

Extract the .toDate to a const instead of repeatedly retrieving it. 
Place outside of the switch. You use it in both paths anyway.